io/www/src/fragment/wcs.js

```diff
@@ -0,0 +1,133 @@
+const { log, logError, fetch } = require('./common.js');
+
+const MAS_ELEMENT_REGEXP = /<[^>]+data-wcs-osi=\\"(?<osi>[^\\]+)\\"[^>]*?>/gm;
+const PROMOCODE_REGEXP =
+    /(?<promo>data-promotion-code=\\"(?<promotionCode>[^\\]+)\\")/;
+
+async function fetchArtifact(osi, promotionCode, wcsContext) {
+    const url = new URL(wcsContext.wcsURL);
+    url.searchParams.set('country', wcsContext.country);
+    url.searchParams.set('locale', wcsContext.locale);
+    url.searchParams.set('landscape', wcsContext.landscape);
+    url.searchParams.set('api_key', wcsContext.context.api_key);
+    url.searchParams.set('language', wcsContext.language);
```

Comment:
setting language always will not work with some UK and perpetual offers, pls see https://github.com/adobecom/milo/blob/stage/libs/features/mas/src/wcs.js#L139

you also miss options.currency, but i am not sure for which use cases its needed.

if future it would be cool to import evergreen mas.js from MAS repo here, and just call resolveWcsOffers method. But I understand its too early for that..

Reply:
here language can't be undefined as i'm setting it strictly from what i do have which is locale, from where i always have a country, and thus always a language being mult or en.
I understand for some offer this will not work, but here not working is that you will get a GB-EN cache entry and a cache miss at consumer side, which i found better than a solution to handle only that case.